api/src/main/java/ai/djl/ndarray/NDArray.java

```diff
@@ -4397,6 +4397,16 @@ default NDArray countNonzero(int axis) {
      */
     NDArrayEx getNDArrayInternal();
 
+    /**
+     * Runs the debug string representation of this {@code NDArray}.
+     *
+     * @return the debug string representation of this {@code NDArray}
+     */
+    default String toDebugString() {
+        int maxSize = Integer.getInteger("ai.djl.nd_display_limit", 0);
```

Comment:
This makes the default maxSize 0, which means it won't show any NDArrays. We should probably have a larger default to allow some to be displayed. Maybe 100 like MxNDarray before?

Reply:
Currently, toString() is called by default in debugger, retrieving the data makes the debugger very slow. The default should be 0. User can manually configure debugger to use toDebugString() when then expend the data view, this will be on-demand and won't impact performance.

We display the content mostly in Jupyter notebook, I'm consider add the property in the IJava kernel to workaround this issue.

Reply:
It would also affect println statements as well. Here is another option. What if we added something like [this check](https://www.rgagnon.com/javadetails/java-detect-if-running-in-debug-mode.html) to see if the JVM is running in debug mode?

Reply:
Now we won't print ndarray content in debugger